resteasy/src/test/scala/com/cerner/beadledom/resteasy/ModuleProcessorSpec.scala

```diff
@@ -0,0 +1,68 @@
+package com.cerner.beadledom.resteasy
+
+import org.jboss.resteasy.spi.{Registry, ResteasyProviderFactory}
+import org.scalatest.{FunSpec, MustMatchers}
+import org.scalatest.mockito.MockitoSugar
+import com.cerner.beadledom.guice.dynamicbindings.AnnotatedModule
+import com.google.inject.Guice
+
+class ModuleProcessorSpec extends FunSpec with MockitoSugar with MustMatchers{
+
+  val registry: Registry = mock[Registry]
+  val providerFactory: ResteasyProviderFactory = mock[ResteasyProviderFactory]
+
+  describe("ModuleProcessor#new") {
+    it("throws a NullPointerException for null registry") {
+      intercept[NullPointerException] {
+        new ModuleProcessor(null, providerFactory)
+      }
+    }
+
+    it("throws a NullPointerException for null provider factory") {
+      intercept[NullPointerException] {
+        new ModuleProcessor(registry, null)
+      }
+    }
+  }
+
+  describe("ModuleProcessor#processInjector") {
+    it("throws a NullPointerException for null injector") {
+      intercept[NullPointerException] {
+        new ModuleProcessor(registry, providerFactory).processInjector(null)
+      }
+    }
+
+    it("excludes providers that have a binding annotation") {
+      val injector = Guice.createInjector(new AnnotatedModule(classOf[TestBindingAnnotation]) {
+        override def configure(): Unit = {
+          bind(classOf[TestJaxRsProvider]).annotatedWith(getBindingAnnotation).toProvider(
```

Comment:
I don't think the call to `toProvider` is needed for this test. Using just `bind(...).annotatedWith(...)` should test the same thing.

Reply:
You can do `.annotatedWith(classOf[TestBindingAnnotation])`.

Reply:
Cannot remove the toProvider because it throws an error.

`No implementation for com.cerner.beadledom.resteasy.TestJaxRsProvider annotated with interface com.cerner.beadledom.resteasy.TestBindingAnnotation was bound.
`